### DISCUSSION TITLE
Radar warning simply doesn't work

### DISCUSSION THREAD

**Discussion by LucasBS1** [osmandapp/OsmAnd#16162] [General]
Hello

Everything is enabled... screen alerts, voice prompts...
But the goddam app won't warn me of a radar !
(Only a small icon on the screen)

Where should I begin to investigate ?

.
![1](https://user-images.githubusercontent.com/86728997/210812997-931d49c8-6ecf-4c3f-b898-41ce32e5d9c0.png)

.

![2](https://user-images.githubusercontent.com/86728997/210813030-95f87e51-171f-45f4-9f8f-9bd2c257ea59.png)

.

![4](https://user-images.githubusercontent.com/86728997/210813040-c80b395a-c2aa-4cc7-90e3-c6608c6b2928.png)

**Comment by mikehgentry** [comment]
You haven't hit 

Settings -> OsmAnd settings -> Uninstall speed cameras

by mistake have you?

**Reply by LucasBS1** [reply]
Thank you for the response

But No

Either I'm blind or this option doesn't exist

![Screenshot_20230112-102336](https://user-images.githubusercontent.com/86728997/212078244-523fe986-0af9-4633-9a0d-b9f935c2cd4b.png)

**Reply by mikehgentry** [reply]
I think you're using an older version, or possibly it's a difference between iOS / Android - I'm on Android 4.3.5

**Reply by LucasBS1** [reply]
Indeed, it's an older version.

I'll update and return here if it's not fixed
Might take some time, as I have to hit the road to test it

Thank you for seeing this

**Reply by mariush444** [reply]
it is not problem with version in my opinion.
I have 4.2.7 and 4.3.5 both sometimes miss radar info.
Sometimes it is bug but sometimes it is a problem with speed camera definition in OSM.
If speedcam is just POI then osmand do nothing with it. If speedcam is realated to road/street then it works OK.
It is just my observation.

**Comment by LucasBS1** [comment]
My problem has changed a little (or a lot)

How do I disable voice commands but keeping the radars ?
If I leave voice enabled, a woman keeps talking
If I disable, the radar warnings go too

![Voice](https://user-images.githubusercontent.com/86728997/219486368-f232e406-b78f-4acf-ad9f-d65b5d3dd6fa.png)

**Reply by mikehgentry** [reply]
I think if you start a navigation then cancel it (or possibly let it finish) that'll kick the voice engine into life, hopefully then it'll announce things if they're selected without giving you directions (since it won't have anywhere to give you directions to)

**Comment by meyverick** [comment]
I'm facing the same two issues; I only want audio and visual alerts for traffic, speed camera and pedestrian and disable voice direction.

Have you managed to enable speed camera alert?
Have you managed to disable voice only for direction?

**Reply by mikehgentry** [reply]
Did you try my suggestion above? https://github.com/osmandapp/OsmAnd/discussions/16162#discussioncomment-5008664

The only other way I'm aware of that you could do it on Android (no idea about iOS) is by editing the javascript to remove the phrases you don't want (or replacing the .oggs, of course). It takes a while, but is pretty trivially easy if you understand anything at all about programming:

1. Download a recorded voice pack you don't use (I used Arabic, since I don't speak Arabic and it's at the top of the list).
2. Download the recorded voice pack for your preferred language.
3. Copy the _tts.js file from your preferred language's directory, and overwrite the one in the language you don't use's directory, without changing the name of the target file. E.g. in the ar (Arabic) directory, the file should be called ar_tts.js, but have the contents of the English one (assuming you want it to be English).
4. Delete the voice directory from the target language, and replace it with the one from your preferred language.
5. Edit the js file, replacing all the phrases you don't want it to say anything for with " ".

You can now just select Arabic (or whatever) in OsmAnd and get your modified English (or whatever) voice pack. Be aware that it'll get overwritten if it's updated, so keep a copy somewhere!

**Reply by LucasBS1** [reply]
Long ago, I attempted to edit en-tts.js to remove anything not related to speed camera

It did not work, even though I carefully removed only what was strictly necessary. There was no sound at all.

I gave up, but here's the path, in MY CASE (depends where the maps are in your phone):

/storage/emulated/0/osmand/voice/en-tts

I edited the one I use (english). Didn't edit a less relevant one, as did Mike above (arab)
Also, in my case, I use text-to-speech. Didn't download the pre-recorded version
